Robot2018/src/org/usfirst/frc/team199/Robot2018/commands/ShiftHighGear.java

```diff
@@ -0,0 +1,47 @@
+package org.usfirst.frc.team199.Robot2018.commands;
+
+import org.usfirst.frc.team199.Robot2018.Robot;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ *
+ */
+public class ShiftHighGear extends Command {
+
+	Timer tim;
+    public ShiftHighGear() {
+        // Use requires() here to declare subsystem dependencies
+        // eg. requires(chassis);
+    }
+
+    // Called just before this Command runs the first time
+    protected void initialize() {
+    	tim = new Timer();
+    	tim.reset();
+    	tim.start();
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    protected void execute() {
+    	Robot.dt.pushGear(true);
+    	SmartDashboard.putBoolean("High Gear", true);
+    }
+
+    // Make this return true when this Command no longer needs to run execute()
+    protected boolean isFinished() {
+        return tim.get() > 0.1;
```

Comment:
Magic numbers like this are usually a sign that something is wrong with the design. In this case, is there a reason that this command needs to end after 0.1 sec? Could you achieve the desired behavior without the magic numbers by creating a "gear shift" subsystem with interruptible commands to shift into high, low, and "neutral"?

Reply:
We had an issue last year with solenoids not activating because the command finished before the pneumatics system could react; magic number issue resolved (moved to SmartDashboard)

Reply:
Putting the number on the SmartDashboard is better but I don't see why the command needs to finish after a particular time at all. Why not just let it run until the user shifts to another gear? That's one less parameter you need on the SD.

Reply:
If this program runs forever, first, there's a possibility that the solenoid might overload (I don't exactly know how the solenoid works internally; this may be blatantly false) and more importantly if another button is pressed that tells the solenoid a different value, then the solenoid will most likely break :( It is easier and also cleaner to have the command terminate after a set time than to search at the beginning of each related command to remove instances of this one from the scheduler.

Reply:
We implemented the time constraint last year after having frenetic solenoids that wouldn't stop switching. I don't remember exactly how we created the problem or if our set up this year will even recreate it, but it's something we can test fairly quickly.

Reply:
Moved to issue #31.